### ISSUE TITLE
vdirsyncer requires an outdated version of python-click

### ISSUE THREAD

**Issue by protist** [pimutils/vdirsyncer#831]
* Your vdirsyncer version: 0.16.8
* If applicable, which server software (and which version) you're using
* Your Python version: python 3.8.3, python-click 7.1.2
* Your operating system: (Arch) Linux

Arch Linux has updated to the latest version of python-click. vdirsyncer no longer launches. Please update vdirsyncer to work with the latest click.

```
$ vdirsyncer -vdebug
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (click 7.1.2 (/usr/lib/python3.8/site-packages), Requirement.parse('click<6.0,>=5.0'), {'vdirsyncer'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/vdirsyncer", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3262, in <module>
    def _initialize_master_working_set():
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3245, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3274, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'click<6.0,>=5.0' distribution was not found and is required by vdirsyncer
```

**Comment by quite** [comment]
Heads up @dvzrv -- hopefully a new working patch release of vdirsyncer is coming up :)

**Comment by dvzrv** [comment]
Ah... my bad. I should've patched that version requirement out. Will bump the package in the meantime.

**Comment by WhyNotHugo** [comment]
Ugh. Current master already drops that limitation, but also had other changes I'd rather not release just yet.

Since Arch's package is already patched, I'd rather just avoid the point release for now and continue working towards 0.18.0 (which drops older Pythons and older stuff and makes maintenance easier).

**Comment by protist** [comment]
Thanks all for getting onto this so quickly.

@dvzrv For future reference, where is the best place to file this? From this [previous comunication](https://bugs.archlinux.org/task/58987), polyzen suggested filing it here (upstream), but I seems like you should have been pinged as well? Should I file it in both places next time?

**Comment by polyzen** [comment]
Both is good if a fix isn't yet available and the downstream package isn't functional (or the issue is "severe" enough, eg. breaks a large number of peoples' workflows). Since the fix was already available in master, to be backported, a ticket downstream is all that was needed.

**Comment by protist** [comment]
Thanks @polyzen. Re-reading the linked comment above, I think I must have misunderstood. I thought your comment downstream was saying not to post in the Arch tracker, only upstream. However, presumably you meant to post in both places (for that particular issue). Thank you for the clarification.

**Comment by nodeg** [comment]
**FYI**: I got the following error during installation, but the installation was successful and I can use vdirsyncer.

```shell
$ pip install vdirsyncer
Defaulting to user installation because normal site-packages is not writeable
Collecting vdirsyncer
  Using cached vdirsyncer-0.16.8-py2.py3-none-any.whl (58 kB)
Collecting click<6.0,>=5.0
  Using cached click-5.1-py2.py3-none-any.whl (65 kB)
Requirement already satisfied: click-log<0.4.0,>=0.3.0 in /home/dom/.local/lib/python3.8/site-packages (from vdirsyncer) (0.3.2)
Requirement already satisfied: atomicwrites>=0.1.7 in /usr/lib/python3.8/site-packages (from vdirsyncer) (1.4.0)
Requirement already satisfied: requests-toolbelt>=0.4.0 in /home/dom/.local/lib/python3.8/site-packages (from vdirsyncer) (0.9.1)
Requirement already satisfied: requests>=2.20.0 in /usr/lib/python3.8/site-packages (from vdirsyncer) (2.24.0)
Requirement already satisfied: click-threading>=0.2 in /home/dom/.local/lib/python3.8/site-packages (from vdirsyncer) (0.4.4)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3.8/site-packages (from requests>=2.20.0->vdirsyncer) (2020.4.5.1)
Requirement already satisfied: chardet>=3.0.2 in /usr/lib/python3.8/site-packages (from requests>=2.20.0->vdirsyncer) (3.0.4)
Requirement already satisfied: idna>=2.5 in /usr/lib/python3.8/site-packages (from requests>=2.20.0->vdirsyncer) (2.10)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/lib/python3.8/site-packages (from requests>=2.20.0->vdirsyncer) (1.25.10)
Installing collected packages: click, vdirsyncer
  Attempting uninstall: click
    Found existing installation: click 7.1.2
    Uninstalling click-7.1.2:
      Successfully uninstalled click-7.1.2
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

black 19.10b0 requires click>=6.5, but you'll have click 5.1 which is incompatible.
Successfully installed click-5.1 click-threading-0.4.4 requests-toolbelt-0.9.1 vdirsyncer-0.16.8
```

**Comment by polyzen** [comment]
Resolved https://github.com/pimutils/vdirsyncer/commit/3eb9ce5ae4320d52e6c876874511ff96a8a45f51

**Comment by WhyNotHugo** [comment]
Fix released in 0.18.0.
